src/main/scala/mesosphere/marathon/plugin/PluginManager.scala

```diff
@@ -0,0 +1,63 @@
+package mesosphere.marathon.plugin
+
+import java.net.{ URL, URLClassLoader }
+import java.util.ServiceLoader
+
+import org.slf4j.{ Logger, LoggerFactory }
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+/**
+  * Instances of this class holds a reference to a plugin of a specific type.
+  * @param classTag the class tag of the specific type
+  * @param plugins all the instances of this plugin
+  * @tparam T the specific type of this plugin.
+  */
+case class PluginReference[T](classTag: ClassTag[T], plugins: Seq[T])
```

Comment:
The Seq here is eager. The iterator returned by the ServiceLoader is lazy. Perhaps we should maintain the lazyness?